### PR TITLE
Feature/formatAmount conversion & wallet.settings

### PR DESCRIPTION
**Pull request by justinkook**
Changes:

### Config.ts

config.wallet.settings schema changed to:
```
wallet: {
    requiredCopayers: number;
    totalCopayers: number;
    spendUnconfirmed: boolean;
    reconnectDelay: number;
    idleDurationMin: number;
    settings: {
      default: WalletSettings;
      btc: WalletSettings;
      bch: WalletSettings;
      eth: WalletSettings;
    };
  };
```

WalletSettings type object is:
```
export interface WalletSettings {
  unitName: string;
  unitToSatoshi: number;
  unitDecimals: number;
  unitCode: string;
  alternativeName: string;
  alternativeIsoCode: string;
  defaultLanguage: string;
  feeLevel: string;
}
```

### tx-format
- formatAmount now requires a coin.
- satToUnit now requires a coin.
> In BWC need to add coin details to common/constants.UNITS

### CODE REVIEW

**Comment by justinkook** [comment]
**In BWC -> common/constants.UNITS**

**There is no bch unit so should we add bch or convert bch to btc in formatAmount?**

File: src/providers/tx-format/tx-format.ts [hunk 1]
```diff
@@ -37,23 +37,24 @@ export class TxFormatProvider {
...
   // TODO: Check return of formatAmount(...), sometimes returns a number and sometimes a string
-  public formatAmount(satoshis: number, fullPrecision?: boolean) {
-    let settings = this.configProvider.get().wallet.settings;
+  public formatAmount(coin: string, satoshis: number, fullPrecision?: boolean) {
+    let { unitCode } = this.configProvider.get().wallet.settings[coin];
 
-    if (settings.unitCode == 'sat') return satoshis;
+    if (unitCode == 'sat') return satoshis;
+    if (unitCode == 'bch') {
+      unitCode = 'btc';
```
Comment (justinkook) [inline src/providers/tx-format/tx-format.ts]: In Bitcore Wallet Client -> Common -> Constants.UNIT there is no bch object

**Comment by bob261002** [comment]
Ya hello justin where is my money???

**Comment by bob261002** [comment]
Thanks bro

Best Regards
BOB AZAHARY BIN HARDI

**Pull request closed without merging**